### PR TITLE
Remove Xamarin

### LINKED ISSUE #4715: [Feature Request] Remove Xamarin targets
### MSAL client type

Public

### Problem statement

Xamarin support ends on May 1, 2024. See [Announcing the Upcoming Deprecation of MSAL.NET for Xamarin and UWP](https://devblogs.microsoft.com/identity/uwp-xamarin-msal-net-deprecation/) for details and links.

### Proposed solution

Remove Xamarin targets, dev apps, and anything else related.

### Alternatives

_No response_

### PR DESCRIPTION
**Pull request by bgavrilMS**
Fixes #4715

Remove Xamarin!!

### CODE REVIEW

File: src/client/Microsoft.Identity.Client/Microsoft.Identity.Client.csproj [hunk 1]
```diff
@@ -8,25 +8,22 @@
...
+    <TargetFrameworkNet6Android>net6.0-android</TargetFrameworkNet6Android>
+    <TargetFrameworkNet6Ios>net6.0-ios15.4</TargetFrameworkNet6Ios>
+    <TargetFrameworkUap>uap10.0.17763</TargetFrameworkUap>
   </PropertyGroup>
 
   <!-- net6.0 (MAUI) -->
   <PropertyGroup Condition="'$(TargetFramework)' == '$(TargetFrameworkNet6Ios)' or '$(TargetFramework)' == '$(TargetFrameworkNet6Mac)' or '$(TargetFramework)' == '$(TargetFrameworkNet6Android)'">
     <DefineConstants>$(DefineConstants);MAUI</DefineConstants>
     <SupportedOSPlatformVersion Condition="'$(TargetFramework)' == '$(TargetFrameworkNet6Ios)'">14.2</SupportedOSPlatformVersion>
-    <SupportedOSPlatformVersion Condition="'$(TargetFramework)' == '$(TargetFrameworkNet6Mac)'">14.0</SupportedOSPlatformVersion>
```
Comment (pmaytak) [inline src/client/Microsoft.Identity.Client/Microsoft.Identity.Client.csproj:23]: There are other references to `TargetFrameworkNet6Mac` in this file and in platform_and_feature_flags.props that can be removed.

File: src/client/Microsoft.Identity.Client/Microsoft.Identity.Client.csproj [hunk 2]
```diff
@@ -238,4 +191,8 @@
   <ItemGroup>
     <PackageReference Include="Microsoft.IdentityModel.Abstractions" />
   </ItemGroup>
+
+  <ItemGroup>
```
Comment (pmaytak) [inline src/client/Microsoft.Identity.Client/Microsoft.Identity.Client.csproj:195]: Remove this? Redundant? Should be already included by default.

File: Directory.Packages.props [hunk 3]
```diff
@@ -29,7 +29,6 @@
     <PackageVersion Include="System.ServiceModel.Security" Version="4.5.3" />
     <PackageVersion Include="System.Xml.XDocument" Version="4.3.0" />
     <PackageVersion Include="System.Xml.XmlDocument" Version="4.3.0" />
-    <PackageVersion Include="Xamarin.Android.Support.CustomTabs" Version="28.0.0.3" />
     <PackageVersion Include="Xamarin.AndroidX.Browser" Version="1.4.0" PrivateAssets="All" />
```
Comment (pmaytak) [inline Directory.Packages.props:32]: Should this be removed too? We add it in net6-android, but I didn't see anywhere where we use this namespace in code.

**Review by pmaytak — CHANGES_REQUESTED** [review]
Left some suggestions.

**Comment by pmaytak** [comment]
@bgavrilMS Does this branch build for you locally? Fails to build for me and installing MAUI workloads also fails (can't find some dependency in the feeds....). :(

**Comment by bgavrilMS** [comment]
> @bgavrilMS Does this branch build for you locally? Fails to build for me and installing MAUI workloads also fails (can't find some dependency in the feeds....). :(

No, it fails for me too due to old .NET SDK. I ended up using net7-android / net7-ios to verify the build etc. 

I was thinking of bumping to net7-*** althoghter, but it's not trivial - there are some APIs which are now obsolete and I don't have a full Android / iOS setup to test. We can revisit this if it causes build issues on the lab.

**Review by pmaytak — APPROVED** [review]
:shipit:

**Review by trwalke — COMMENTED** [review]
No Longer needed?
https://github.com/AzureAD/microsoft-authentication-library-for-dotnet/blob/4a8c4ddae8a36d811d4ebf55af30d0feba03577d/LibsAndSamples.sln#L138

**Review by trwalke — COMMENTED** [review]
I dont think we still need LibsMacOS.sln

**Review by trwalke — COMMENTED** [review]
No longer need template-build-android-automation.yaml

File: build/platform_and_feature_flags.props [hunk 4]
```diff
@@ -9,16 +9,16 @@
...
   </PropertyGroup>
-  <PropertyGroup Condition=" $(TargetFramework.StartsWith('MonoAndroid')) Or '$(TargetFramework)' == '$(TargetFrameworkNet6Android)'">
-    <DefineConstants>$(DefineConstants);ANDROID;SUPPORTS_BROKER;IS_XAMARIN_OR_UWP</DefineConstants>
+  <PropertyGroup Condition="'$(TargetFramework)' == '$(TargetFrameworkNet6Android)'">
+    <DefineConstants>$(DefineConstants);ANDROID;SUPPORTS_BROKER</DefineConstants>
   </PropertyGroup>
   <PropertyGroup Condition="'$(TargetFramework)' == '$(TargetFrameworkNetDesktop462)'">
     <DefineConstants>$(DefineConstants);SUPPORTS_BROKER;SUPPORTS_CONFIDENTIAL_CLIENT;SUPPORTS_CUSTOM_CACHE;SUPPORTS_WIN32</DefineConstants>
   </PropertyGroup>
-  <PropertyGroup Condition="'$(TargetFramework)' == '$(TargetFrameworkIos)' Or '$(TargetFramework)' == '$(TargetFrameworkNet6Ios)' Or '$(TargetFramework)' == '$(TargetFrameworkNet6Mac)'">
```
Comment (neha-bhargava) [inline build/platform_and_feature_flags.props:20]: Does `TargetFrameworkNet6Mac` also needs to be removed?

File: Directory.Packages.props [hunk 5]
```diff
@@ -4,7 +4,7 @@
     <!-- Version of the Microsoft.Identity.Client.NativeInterop package. -->
     <MSALRuntimeNativeInteropVersion>0.16.0</MSALRuntimeNativeInteropVersion>
     <!-- Version of MSAL if not defined by the CI-->
-    <MsalInternalVersion>4.51.0</MsalInternalVersion>
+    <MsalInternalVersion>4.61.0</MsalInternalVersion>
```
Comment (gladjohn) [inline Directory.Packages.props:7]: undo?

**Review by gladjohn — APPROVED** [review]
looks good. Can we also delete the Automation framework / subscription we used to have to run Android and IOS tests? Do not remember the name now just making sure we do the clean up of that if not required anymore

**Comment by bgavrilMS** [comment]
> looks good. Can we also delete the Automation framework / subscription we used to have to run Android and IOS tests? Do not remember the name now just making sure we do the clean up of that if not required anymore

For sure. I don't know where it is though... @trwalke ?

**Review by trwalke — APPROVED** [review]

**Review by neha-bhargava — APPROVED** [review]

**Review by pmaytak — APPROVED** [review]

**Pull request merged**